doc/event-alarm-framework/event-alarm-framework.md

```diff
@@ -0,0 +1,1265 @@
+Id:                  2 
+Action:              RAISE 
+Severity:            CRITICAL 
+Type:                DUMMY_ALARM 
+Timestamp            2021-05-19T21:39:31.622Z 
+Description:         :- signalHandler: Raising simulated alarm 
+Source:              simulation 
+
+sonic# show event from 2 to 5
+----------------------------------------------------------------------------------------------------------------------------
+Id           Action          Severity   Name                           Timestamp                   Description                                                  
+----------------------------------------------------------------------------------------------------------------------------
+2            RAISE           CRITICAL   DUMMY_ALARM                    2021-05-19T21:39:31.622Z    :- signalHandler: Raising simulated alarm         
+3            CLEAR           CRITICAL   DUMMY_ALARM                    2021-05-19T21:42:34.371Z    :- signalHandler: Clearing simulated alarm        
+4            RAISE           CRITICAL   DUMMY_ALARM                    2021-05-19T21:46:14.371Z    :- signalHandler: Raising simulated alarm         
+5            ACKNOWLEDGE     CRITICAL   DUMMY_ALARM                    2021-05-19T21:48:05.845Z    Alarm id 4 ACKNOWLEDGE.
+
+sonic# show event start 2021-05-19T21:39:31.622Z end 2021-05-19T21:46:14.371Z
+----------------------------------------------------------------------------------------------------------------------------
+Id           Action          Severity   Name                           Timestamp                   Description                                                  
+----------------------------------------------------------------------------------------------------------------------------
+3            CLEAR           CRITICAL   DUMMY_ALARM                    2021-05-19T21:42:34.371Z    :- signalHandler: Clearing simulated alarm 
+
+sonic# show alarm [ acknowledged | all | detail | summary | severity <sev> | id <seq-id> | start <from-ts> end <to-ts> | recent <5min|1hr|1day> | from <from-seq> to <to-seq> ]
+
+'show alarm' command would display all the *active* alarm records in ALARM table. Acknowledged alarms wont be shown here.
+
+sonic# show alarm
+----------------------------------------------------------------------------------------------------------------------------
+Id           Severity   Name                           Timestamp                   Description                                                  
+----------------------------------------------------------------------------------------------------------------------------
+14           WARNING    TEMPERATURE_EXCEEDED           2021-05-20T00:47:52.992Z    :- temperatureCrossedThreshold: Current temperature of sensor/2 is 76 degrees         
+16           WARNING    PSU_FAULT                      2021-05-20T02:16:42.611Z    :- /psu/2 has experienced a fault 
+
+sonic# show alarm all
```

Comment:
Not clear what is the difference between show alarm without any additional input and all.
in the example there is only one alarm that is missing but i could not understand why 16 is missing with show alarm is executed. Can you please share the logic? Maybe it is somewhere else in the doc?

Reply:
This is mentioned above these commands: "'show alarm' command would display all the *active* alarm records in ALARM table. Acknowledged alarms wont be shown here.",

As mentioned, 'show alarm' wont display acknowledged alarms. 

'show alarm all' displays acknowledged alarms as well.

Reply:
i think it is better to define the CLI in an opposite way
'show alarm' is for all
'show alarm active'

Reply:
It is 'show alarm' command that operator uses quite often. 
IMHO operator would want to see "active" alarms and don't want those alarms he/she finds not relevant and acknowledged.
If at all operator wants to see all, 'all' option is there.